Uds-1285

scott-williams-az opened the pull request:
### Description
Webpack changes for unity-bootstrap-theme build
- generate files needed for gh-pages
- copy sort storybook from bs4 to bs5
- add #html-root to storybook not using a template

### Links

- [JIRA ticket](https://asudev.jira.com/browse/UDS-1285)

### Checklist

- [x] Unity project successfully builds from root `yarn install` & `yarn build`
- [x] Commits do not contain multiple scopes
- [x] Add/updated examples

scott-williams-az committed:
fix(unity-bootstrap-theme): build-storybook

scott-williams-az committed:
fix(unity-bootstrap-theme):sort storybook

scott-williams-az committed:
fix(unity-bootstrap-theme):storybook HTML viewsource code

scott-williams-az committed:
fix(unity-bootstrap-theme): swap file names

davidornelas11 committed:
fix: added new css in storybook index page

maycomayco commented on packages/unity-bootstrap-theme/webpack.config.js:
```diff
@@ -30,7 +30,7 @@ const jsConfig = {
   entry: {
     "global-header": [path.resolve(paths.js, "global-header.js")],
-    "googleAnalytics": path.resolve(paths.js, "googleAnalytics.js"),
+    "global-header-datalayer": path.resolve(paths.js, "googleAnalytics.js"),
```
The google analytics script is to be used for the whole package and not only for the header, maybe the name is not the right one.

scott-williams-az replied:
I was looking to see where googleAnalytics was added https://github.com/ASU/asu-unity-stack/blob/dev/packages/bootstrap4-theme/gulpfile.js#L163-L166 But I think you are right that I got something wrong here,

mlsamuelson replied:
Agreed.

mlsamuelson replied:
Unresolving this.

Curious about the latest commit here: https://github.com/ASU/asu-unity-stack/pull/1034/commits/65443138f2d643e2ab3917a6da56e23621f8a205 We want to keep the global-header.js and googleAnalytics.js separate.  The concern that 
@maycomayco flagged at https://github.com/ASU/asu-unity-stack/pull/1034#pullrequestreview-1419504930 and which I seconded was the file naming - we should keep the file named googleAnalytics.js  and only containing the dataLayer code. Someone may be using the base theme but not the header, and they'll need the GA code.

mlsamuelson committed:
docs: Update bootstrap4-theme content wording

davidornelas11 committed:
fix: changed references from bs4 to unity bs5

scott-williams-az committed:
fix(webpack): fix webpack global-header-datalayer output

davidornelas11 commented on server/storybook-index-generator.js:
```diff
@@ -17,24 +17,21 @@ module.exports = generateHTML = packages => `
-    <script type="text/javascript" src="./@asu/bootstrap4-theme/assets/fontawesome/js/all.js"></script>
+    <script type="text/javascript" src="./@asu/unity-bootstrap-theme/js/bootstrap.bundle.min.js"></script>
+    <script type="text/javascript" src="./@asu/unity-bootstrap-theme/js/global-header.js"></script>
+    <script type="text/javascript" src="./@asu/unity-bootstrap-theme/js/global-header-datalayer.js"></script>
```
@mlsamuelson @scott-williams-az @maycomayco Wasn't sure where we left off on this so I kept it at the global-header-dataLayer name. Is this correct?

maycomayco replied:
Because the latest things discussed this comment is outdated

mlsamuelson committed:
docs(unity-bootstrap-theme): fix SB docs sorting and title issues

scott-williams-az commented on packages/unity-bootstrap-theme/.storybook/preview.js:
```diff
@@ -30,6 +30,7 @@ export const parameters = {
             'Navbar options',
```
Scroll up to see 'Components', can probably be removed

mlsamuelson replied:
Yes. It can. Not going to consider it blocking for this PR though.

mlsamuelson replied:
Was testing locally so I took care of this.

mlsamuelson committed:
build: gtm docs updates. remove asuthemes

Updates and refactors GTM docs and static site layout. Removes ASUThemes page. Updates header docs
in Storybook.

mlsamuelson committed:
build: fix find and replace error

mlsamuelson committed:
build: fix grammar

maycomayco commented on packages/unity-bootstrap-theme/src/js/global-header-datalayer.js:
This file was renamed but there is a reference with the old name that was not renamed here: `global-header.examples.stories.js`

mlsamuelson committed:
Merge pull request #1037 from ASU/uds-1285-gtm-datalayer

Uds 1285 gtm datalayer

scott-williams-az committed:
refactor(unity-bootstrap-theme): refactor GA code and header scroll

scott-williams-az committed:
refactor(unity-bootstrap-theme): include scripts in static index

davidornelas11 committed:
Merge pull request #1038 from ASU/UDS-1285-ga-and-scroll

refactor(unity-bootstrap-theme): refactor GA code and header scroll

scott-williams-az committed:
chore(unity-bootstrap-theme): dataLayer example is readable

Removed shorthand code and made data layer example more clear for others to implement

scott-williams-az committed:
chore(unity-bootstrap-theme): gA code should not be duplicated

mlsamuelson committed:
chore(unity-bootstrap-theme): remove cruft from storybook sort

scott-williams-az committed:
chore(unity-bootstrap-theme): remove duplication of js files for static site

scott-williams-az committed:
chore(unity-bootstrap-theme): js files work in storybook and static site

davidornelas11 committed:
Merge pull request #1039 from ASU/UDS-1285-ga-and-scroll2

chore(unity-bootstrap-theme): dataLayer example is readable

davidornelas11 committed:
chore: edited njk templates and added readme info for datalayer and header

davidornelas11 committed:
Update index.njk

scott-williams-az committed:
chore(unity-bootstrap-theme): remove some comments

scott-williams-az committed:
chore(unity-bootstrap-theme): rename class to className in JSX code

scott-williams-az committed:
chore(unity-bootstrap-theme): update style object for JSX

scott-williams-az committed:
chore(unity-bootstrap-theme): remove React case sensitive warning

davidornelas11 committed:
Merge pull request #1043 from ASU/uds-1285-example-readme

chore: edited njk templates and added readme info for datalayer and header

mlsamuelson committed:
Merge pull request #1045 from ASU/UDS-1285-JSX

Uds 1285 jsx

mlsamuelson reviewed (approved):
This one builds and looks good to me. Mostly comprised of other PRs already vetted.
